linera-service/src/proxy.rs

```diff
@@ -57,8 +65,33 @@ impl Proxy {
     }
 
     /// Constructs and configures the [`Proxy`] given [`ProxyOptions`].
-    fn from_options(options: ProxyOptions) -> Result<Self> {
-        let config = ValidatorServerConfig::read(&options.config_path)?;
+    async fn from_options(options: ProxyOptions) -> Result<Self> {
+        let mut tmp_config = ValidatorServerConfig::read(&options.config_path)?;
+        if options.kube {
+            let pod_name_to_ip = fetch_pod_ips().await?;
+
+            // Assumes that each service will have just one pod. Since we have one service per shard, the first pod will always have
+            // -0 at the end of it for servers and some hash for the validators. So we check the prefix and assume only one entry
+            // will have that prefix.
+            tmp_config.validator.network.host =
+                find_value_with_key_prefix(&pod_name_to_ip, &tmp_config.validator.network.host)
+                    .expect("All hosts should be a prefix of a pod name!")
+                    .clone();
+            tmp_config.internal_network.host =
+                find_value_with_key_prefix(&pod_name_to_ip, &tmp_config.internal_network.host)
+                    .expect("All hosts should be a prefix of a pod name!")
+                    .clone();
+            for shard in &mut tmp_config.internal_network.shards {
+                shard.host = find_value_with_key_prefix(&pod_name_to_ip, &shard.host)
+                    .expect("All hosts should be a prefix of a pod name!")
+                    .clone();
+                shard.metrics_host =
+                    find_value_with_key_prefix(&pod_name_to_ip, &shard.metrics_host)
+                        .expect("All hosts should be a prefix of a pod name!")
+                        .clone();
+            }
+        }
+        let config = tmp_config;
```

Comment:
Not sure if this is acceptable Rust or not, feels wrong ha
Basically creating a temporary mutable variable, altering it, then moving it to an immutable one